### ISSUE TITLE
xunit.methodDisplay appSetting of "method" not respected in rc1-build1030

### ISSUE THREAD

**Issue by MichaelFStarke** [xunit/visualstudio.xunit#43]
Per issue #40, the app settings key should be available in RC1, but trying to override the default of "classAndMethod" to get the behavior of "method" is not respected.

Environment:
VS2012 Ultimate, (V11.0.61030.00 Update4)
xunit.runner.visualstudio 2.0.0-rc1-build1030 (Prerelease)
&lt;appSettings&gt;
  &lt;add key="xunit.methodDisplay" value="method" /&gt;
&lt;/appSettings&gt;

**Comment by MichaelFStarke** [comment]
I have validated that this is not an artifact of my existing solution, by isolating it in a brand new minimal solution.

**Comment by bradwilson** [comment]
Can you provide a repro solution? We use this feature extensively and it seems to be working fine here.

**Comment by MichaelFStarke** [comment]
I would be happy to, but GitHub doesn't seem to want to allow me to attach anything but an image (gif/png/etc.) to this issue.  I have the whole solution .ZIPped, how would you like me to submit it?

**Comment by bradwilson** [comment]
You could create a Github project and upload, but it just occurred to me: are you using v1.x or v2.x? The app settings are by-and-large only supported in v2.x tests.

**Comment by MichaelFStarke** [comment]
I don't have it on hand, but as I recall, my repro solution has runner v2 running v1 tests.  I'm stuck on v1 tests because the .Should library we use hasn't been updated to Xunit v2.

Doesn't the appSetting apply to the runner?  Why is the version used by the tests relevant?  I haven't looked at the code, I'm just curious.

**Comment by bradwilson** [comment]
The runner does not do things like declaring the display name of a unit test. The unit test library is responsible for that. Most of the options are merely read by the runner and handed to the unit test framework to be interpreted.

The one exception is multi-assembly parallelization. That option is purely in the control of the runner and outside the scope of the configuration option.

I've updated the configuration page to reflect this: http://xunit.github.io/docs/configuration-files.html

**Comment by MichaelFStarke** [comment]
But something changed...  I was using the v2 runner for a period of time (after the VSIX runner was deprecated) against v1 tests and seeing the "method" behavior (not "classAndMethod" as it is now).  Now, I am left in a rather untenable situation in that my options are to "upgrade" to prerelease software, stand pat and have test names that are unreadable and indistinguishable from each other due to the namespace/class name prefixes, or manually go through all of my tests and add display names to the fact attributes...

**Comment by bradwilson** [comment]
Yeah, unfortunately, you are a victim of me fixing inconsistency.

When we first shipped the VS extension, we shortened the names (sometimes in a buggy, inappropriate way, BTW) because that's what MSTest did, and their sample code for us did the same thing. However, it was the only runner that did this, and as already mentioned, it had bugs.

The appropriate support belonged in every test runner, and the default is what the default was for all runners -- except Visual Studio. So unfortunately, you're the unlucky recipient of the bug fix, with no way to opt into the old behavior on v1.x.

Sorry about that. :(
